staging/src/k8s.io/legacy-cloud-providers/azure/azure_routes.go

```diff
@@ -184,6 +184,22 @@ func (az *Cloud) CreateRoute(ctx context.Context, clusterName string, nameHint s
 		},
 	}
 
+	actualRoute, err := az.RoutesClient.Get(ctx, az.RouteTableResourceGroup, az.RouteTableName, routeName)
+	if err != nil {
+		klog.V(3).Infof("CreateRoute: creating route: failed(GetRoute) instance=%q with error=%v", kubeRoute.TargetNode, err)
+		return err
+	}
+
+	if actualRoute.RoutePropertiesFormat != nil &&
+		actualRoute.RoutePropertiesFormat.AddressPrefix != nil &&
+		*actualRoute.RoutePropertiesFormat.AddressPrefix == kubeRoute.DestinationCIDR &&
```

Comment:
use `to.String()` is easier, and use `strings.EqualFold()` for case insentive, e.g.

e.g. `strings.EqualFold(to.String(actualRoute.RoutePropertiesFormat.AddressPrefix), kubeRoute.DestinationCIDR)`